src/main/java/nl/tudelft/dnainator/graph/impl/Neo4jQuery.java

```diff
@@ -0,0 +1,96 @@
+package nl.tudelft.dnainator.graph.impl;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.ResourceIterator;
+import org.neo4j.graphdb.Result;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.helpers.collection.IteratorUtil;
+
+import nl.tudelft.dnainator.core.SequenceNode;
+import nl.tudelft.dnainator.graph.GraphQueryDescription;
+
+/**
+ * A useful class for creating and executing a query on a Neo4j
+ * database using a {@link GraphQueryDescription}.
+ */
+public class Neo4jQuery {
+	private String cypherQuery;
+	private boolean multipleConditions = false;
+	private Map<String, Object> parameters;
+	private GraphQueryDescription description;
+
+	/**
+	 * Create a new query suitable for Neo4j from the given description.
+	 * @param qd the query description to use for constructing the query.
+	 */
+	public Neo4jQuery(GraphQueryDescription qd) {
+		this.description = qd;
+		buildQuery();
+	}
+
+	private void addCondition(StringBuilder sb, String c) {
```

Comment:
Is it possible to check for multiple conditions in another way?
For example by not storing it in a state?